local_planner/src/nodes/planner_functions.cpp

```diff
@@ -10,19 +10,26 @@ namespace avoidance {
 
 // trim the point cloud so that only points inside the bounding box are
 // considered
-void filterPointCloud(
-    pcl::PointCloud<pcl::PointXYZ>& cropped_cloud,
+void processPointcloud(
+    pcl::PointCloud<pcl::PointXYZI>& final_cloud,
     Eigen::Vector3f& closest_point, float& distance_to_closest_point,
     int& counter_backoff,
     const std::vector<pcl::PointCloud<pcl::PointXYZ>>& complete_cloud,
-    int min_cloud_size, float min_dist_backoff, Box histogram_box,
-    const Eigen::Vector3f& position, float min_realsense_dist) {
-  cropped_cloud.points.clear();
-  cropped_cloud.width = 0;
+    float min_dist_backoff, Box histogram_box, const Eigen::Vector3f& position,
+    float min_realsense_dist, int max_age) {
+  pcl::PointCloud<pcl::PointXYZI> old_cloud = final_cloud;
+
+  final_cloud.points.clear();
+  final_cloud.width = 0;
```

Comment:
could you reserve the size to the maximum it could be? I think that would be faster than dynamic reallocation

Reply:
Agreed, `final_cloud.points.reserve(histogram.rows() * histogram.cols());`

Repeatedly reserving (and then destroying) the exact same sized chunk of memory also makes things easier for the memory allocator to re-use memory.